docs/concepts.md

```diff
@@ -0,0 +1,123 @@
+# Concepts
+
+PrairieLearn is extremely flexible and powerful, but this flexibility can be overwhelming at first. This document provides an overview of the key concepts in PrairieLearn, and how they fit together.
+
+## High-level overview
+
+A [**Course**](./course.md) in PrairieLearn is composed of **Course Instances**, **Questions**, and **Custom Elements**. Each [course instance](./courseInstance.md) is an offering of a course in a particular semester/term. For example, a course "CS 225" might have instances "CS 225 Spring 2022" and "CS 225 Fall 2022". A course instance contains all the **Assessments** for that particular term. The course itself contains the questions and custom elements that are shared across all instances. You can think of the course as a bank of questions and elements, that are then used to create assessments for each instance of the course.
+
+=== "Concept Map"
+
+    ```d2
+    --8<-- "docs/diagrams/concept-map.d2"
+    ```
+
+=== "Example for the `PrairieLearn 101` course"
+
+    ```d2
+    --8<-- "docs/diagrams/concept-map-example.d2"
+    ```
+
+### Questions
+
+[**Questions**](./question.md) are all independent, and can be graded [using Python](./question.md), or using an [external code autograder](./externalGrading.md). They are written in HTML (as a [Mustache template](https://mustache.github.io/mustache.5.html)) and generated / graded using Python.
```

Comment:
Workshopping this a bit; feel free to make further tweaks. I was trying to remove the double mention of "gradable with Python" and made additional changes from there.

```suggestion
[**Questions**](./question.md) are all independent, and a given question can be used on many different assessment across many different course instances. They are written in HTML (as a [Mustache template](https://mustache.github.io/mustache.5.html)) and use [elements](./elements.md) to accept student input. They can generate random parameters and grade themselves with Python code. Student code submissions can also be graded with an [external code autograder](./externalGrading.md). 
```

Reply:
I have some tweaks in the latest version.